### PR TITLE
feat: add endpoint to handle an election jurisdictions file

### LINKED ISSUE #303: API call to upload set of jurisdictions and admins
- create/replace the jurisdictions
- create the users and mappings to jurisdictions

Likely a

`POST /election/<election_id>/jurisdictions_file`

or something like it.

### PR DESCRIPTION
**Pull request by eventualbuddha**
~This required a bit of a refactor in order to avoid cycles. Mainly, the application routes have been moved to a file called `routes.py` and a bunch of utility functions were moved to their own files.~

I managed to simply change the order of imports to avoid the cycles. That makes this PR more focused. Will follow up with a refactor later.

Closes #303

### CODE REVIEW

**Comment by eventualbuddha** [comment]
@MorganLove can you take a look at this from an API perspective? Seem alright to you?

File: tests/test_app_upload_jurisdictions_file.py [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+        })
+    assert rv.status_code == 200
+    assert json.loads(rv.data) == {'status': 'ok'}
+
+    election = Election.query.filter_by(id=election_id).one()
+    assert [j.name for j in election.jurisdictions] == ['J1', 'J2']
+
+    for jurisdiction in election.jurisdictions:
+        assert [a.user.email
+                for a in jurisdiction.jurisdiction_administrations] == ['a1@example.com']
```
Comment (jonahkagan) [inline tests/test_app_upload_jurisdictions_file.py]: This is officially the best tested feature in the codebase.

File: tests/test_app_upload_jurisdictions_file.py [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+    app.config['TESTING'] = True
+    client = app.test_client()
+
+    with app.app_context():
+        db.drop_all()
+        db.create_all()
+
+    yield client
+
+    db.session.commit()
```
Comment (jonahkagan) [inline tests/test_app_upload_jurisdictions_file.py]: are `post_json` and `client` copy-pasted from `test_app.py`? I imported them in `test_report.py`. Do you think one way or the other is better? Or should we pull them out into a helper module?

File: arlo_server/__init__.py [hunk 3]
```diff
@@ -315,6 +316,55 @@ def election_new():
     return jsonify(electionId=election_id)
 
 
+@app.route('/election/<election_id>/jurisdictions_file', methods=["GET"])
```
Comment (jonahkagan) [inline arlo_server/__init__.py]: I'm still wrapping my head around the API design a bit. Wondering if this should be part of the `/election/{id}/jurisdictions` API or if it's a separate API like you have it here.

File: tests/test_app_upload_jurisdictions_file.py (hunk 2 above)
Comment (eventualbuddha) [reply tests/test_app_upload_jurisdictions_file.py]: They were just copied, yeah. We could put them in `tests/utils.py` or something.

File: tests/test_app_upload_jurisdictions_file.py (hunk 1 above)
Comment (eventualbuddha) [reply tests/test_app_upload_jurisdictions_file.py]: 😂

File: arlo_server/__init__.py (hunk 3 above)
Comment (eventualbuddha) [reply arlo_server/__init__.py]: I mostly just followed @benadida's suggestion from #303, though I think having it in `…/jurisdictions` could be a little confusing. We've been discussing with @mcchilders that elections officials have a mental model that includes "files" for things like the list of jurisdictions being the source of truth, and so we're trying to model that here.

**Comment by eventualbuddha** [comment]
Alright, I'm merging so we can move ahead. @MorganLove, I'm happy to make changes in a follow up PR if it makes client integration easier.

**Pull request merged**